src/charm.py

```diff
@@ -353,6 +403,41 @@ def _get_tls_config(self) -> dict:
             }
         }
 
+    def _get_tracing_config(self) -> dict:
+        """Return dictionary with opentelemetry configuration if available."""
+        # ref: https://doc.traefik.io/traefik/ma<woke forced me to add this>ster/observability/tracing/opentelemetry/
+        if not self._is_tracing_enabled():
+            logger.error("tracing not enabled")
+            return {}
+
+        # traefik supports http and grpc
+        if addr := self._tracing.otlp_grpc_endpoint:
+            grpc = True
+        elif addr := self._tracing.otlp_http_endpoint:
+            grpc = False
+        else:
+            logger.error(
+                "tracing integration is active but none of the "
+                "protocols traefik supports is available."
+            )
+            return {}
+
+        otlp_cfg: Dict[str, Any] = {"address": addr}
+        if self._is_tls_enabled():
+            # todo: we have an option to use CA or to use CERT+KEY authentication.
+            #  what is the difference and which one is better?
+            otlp_cfg["ca"] = _CA_CERT_PATH
+        else:
+            otlp_cfg["insecure"] = True
```

Comment:
Not sure what this `else` means here. Who is the server and who is the client?

I wonder if:
- we should always set "insecure" to True
- we shouldn't set "ca" because we assume the other side (or us) already has it from `update-ca-certificates`

Reply:
setting insecure to True is an error if we also present "ca"
"ca" (or cert+key) are required if insecure=False (the default).

Reply:
that's not in place, not that I know of. is that part of the mTLS story?

Reply:
It's part of the TLS story. @lucabello has 2 merged examples at this point?